source/jquery.flot.drawSeries.js

```diff
@@ -23,124 +23,160 @@ This plugin is used by flot for drawing lines, plots, bars or area.
                     }
 
                     x1 = (axisy.max - y1) / (y2 - y1) * (x2 - x1) + x1;
                     y1 = axisy.max;
                 } else if (y2 >= y1 && y2 > axisy.max) {
                     if (y1 > axisy.max) {
-                        continue;
+                        return true;
                     }
 
                     x2 = (axisy.max - y1) / (y2 - y1) * (x2 - x1) + x1;
                     y2 = axisy.max;
                 }
+            };
 
-                // clip with xmin
+            var handleXMinClipping = function () {
                 if (x1 <= x2 && x1 < axisx.min) {
                     if (x2 < axisx.min) {
-                        continue;
+                        return true;
                     }
 
                     y1 = (axisx.min - x1) / (x2 - x1) * (y2 - y1) + y1;
                     x1 = axisx.min;
                 } else if (x2 <= x1 && x2 < axisx.min) {
                     if (x1 < axisx.min) {
-                        continue;
+                        return true;
                     }
 
                     y2 = (axisx.min - x1) / (x2 - x1) * (y2 - y1) + y1;
                     x2 = axisx.min;
                 }
+            };
 
-                // clip with xmax
+            var handleXMaxClipping = function () {
                 if (x1 >= x2 && x1 > axisx.max) {
                     if (x2 > axisx.max) {
-                        continue;
+                        return true;
                     }
 
                     y1 = (axisx.max - x1) / (x2 - x1) * (y2 - y1) + y1;
                     x1 = axisx.max;
                 } else if (x2 >= x1 && x2 > axisx.max) {
                     if (x1 > axisx.max) {
-                        continue;
+                        return true;
                     }
 
                     y2 = (axisx.max - x1) / (x2 - x1) * (y2 - y1) + y1;
                     x2 = axisx.max;
                 }
+            };
 
+            var drawLine = function () {
                 if (x1 !== prevx || y1 !== prevy) {
                     ctx.moveTo(axisx.p2c(x1) + xoffset, axisy.p2c(y1) + yoffset);
                 }
 
                 prevx = x2;
                 prevy = y2;
                 ctx.lineTo(axisx.p2c(x2) + xoffset, axisy.p2c(y2) + yoffset);
+            };
+
+            ctx.beginPath();
+            for (i = ps; i < points.length; i += ps) {
+                initPoints(i);
+
+                if (x1 === null || x2 === null) {
+                    mx = null;
+                    my = null;
+                    continue;
+                }
+
+                if (isNaN(x1) || isNaN(x2) || isNaN(y1) || isNaN(y2)) {
+                    prevx = null;
+                    prevy = null;
+                    continue;
+                }
+
+                if (steps && handleSteps()) {
+                    // Subtract pointsize from i to have current point p1 handled again if middle point was present.
+                    i -= ps;
```

Comment:
Isn't this only necessary/valid if a middle point was created?

Reply:
Yes, it is only necessary if the middle point was created, and this is what it does. If you check the handleStep, you can see that I've added a return true only when there was a middle point.
<img width="764" alt="image" src="https://user-images.githubusercontent.com/26199384/213580086-47f341a8-ccc8-456d-8d6a-2fd6c5ce037f.png">

Reply:
Ah crap, sorry, I totally missed that, though I do like your suggested change to make it more clear.

Reply:
No problem :) I will do that in few minutes.